src/pages/UserInfo/index.js

```diff
@@ -3,48 +3,142 @@ import logo from '../../images/logo.png';
 import Frame from '../../components/Frame';
 import Button from '../../components/Button';
 import Img from '../../components/Img';
+import MotoInfo from '../../components/MotoInfo';
+import TowInfo from '../../components/TowInfo';
 import { StyledInput, Container } from '../../components/StyledInput/index';
-import { Select, Label } from '../../components/StyledSelect/index';
 import { StyledLink, StyledFieldset } from "./styles";
+import axios from 'axios';
 
 
 class UserInfo extends React.Component {
+
   state = {
     name: '',
-    lastName: '',
     phoneNum:'',
     brand:'',
     cc:'',
     capacity:'',
     plateNum:'',
     type:'',
+    weight:'',
     photo: '',
     vehiPhoto: '',
+    vehicleType:'',
+    edit: false,
   };
-
+  
   handleChange = (event) => {
     const { name, value } = event.target;
     this.setState({
       [name]: value,
     });
   };
 
-  sendInfo = (event) => {
+  handleEdit = (e) => {
+    this.setState({
+      edit: e.target.type === 'checkbox' ? e.target.checked : e.target.value,
+    })
+  }
+
+  async componentDidMount() {
+    const userType = localStorage.getItem('userType');
+    const email = localStorage.getItem('email');
+    const { data } = await axios({
+      method: 'GET',
+      baseURL:process.env.REACT_APP_SERVER_URL,
+      url: `/${userType}s?email=${email}`
+    })
+
+    if (userType === 'user') {
+      const { name, phoneNum } = data.users[0];
+      this.setState({ vehicleType: 'Moto', name, phoneNum })
+    } else {
+      const { name, phoneNum } = data.suppliers[0];
+      this.setState({ vehicleType: 'Grúa', name, phoneNum })
+    }
+  }
+
+  eraseUser = async(event) => {
+    event.preventDefault();
+    const token = localStorage.getItem('token');
+    const userType = localStorage.getItem('userType');
+
+    await axios({
+      method: 'DELETE',
+      baseURL:process.env.REACT_APP_SERVER_URL,
+      url: `/${userType}s`,
+      headers: {
+        Authorization: `Bearer ${token}`,
+      } 
+    })
+    this.setState({
+      name: '',
+      phoneNum:'',
+    })
+    this.props.history.push(`/login/`);
+    localStorage.clear();
+  };
+
+  sendInfo = async(event) => {
     event.preventDefault();
+    const userType = localStorage.getItem('userType');
+    const token = localStorage.getItem('token');
+    if (this.state.edit) {
+      const { data } = await axios({
+        method: 'PUT',
+        baseURL:process.env.REACT_APP_SERVER_URL,
+        url: `/${userType}s`,
+        data: {
+          name: this.state.name,
+          phoneNum: this.state.phoneNum,
+        },
+        headers: {
+          Authorization: `Bearer ${token}`,
+        } 
+      })
+      alert(data.message)
+    }
```

Comment:
Acá falta un else. Cuando this.state.edit es true se envía primero el PUT y luego también se envía el POST

Reply:
Hecho, además se le adiciona el método PUT para actualizar el vehículo